cmd/kubeadm/app/phases/addons/addons_test.go

```diff
@@ -86,3 +92,71 @@ func TestCompileManifests(t *testing.T) {
 		}
 	}
 }
+
+// This unit test exists because tolerations weren't decoded correctly by kuberuntime.DecodeInto
```

Comment:
cc @caesarxuchao @kubernetes/sig-api-machinery-bugs 
I think I reported to @caesarxuchao about the problem in the v1.6 cycle...
Basically the decoder didn't recognize the Tolerations field at all, although the type was added to the code. This was in the end of the cycle, and without these tolerations `kubeadm init` doesn't work properly. So I added the Tolerations manually in the code instead, because I couldn't rely on the Decoder.

This unit test is there to make sure that regression (whatever it was) doesn't happen again.
I don't an addition of an unit test is something to block this PR on though...